centreon/src/Core/Dashboard/Infrastructure/Repository/DbReadDashboardPerformanceMetricRepository.php

```diff
@@ -0,0 +1,525 @@
+                if (! array_key_exists($record['service_id'], $metricsInformation)) {
+                    $metricsInformation[$record['service_id']] = [
+                        'service_id' => $record['service_id'],
+                        'resource_name' => $record['resource_name'],
+                        'metrics' => [],
+                    ];
+                }
+                $metricsInformation[$record['service_id']]['metrics'][] = new PerformanceMetric(
+                    $record['metric_id'],
+                    $record['metric_name'],
+                    $record['unit_name']
+                );
+            }
+            foreach ($metricsInformation as $information) {
+                $resourceMetrics[] = new ResourceMetric(
+                    $information['service_id'],
+                    $information['resource_name'],
+                    $information['metrics']
+                );
+            }
+        }
+
+        return $resourceMetrics;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function FindByRequestParametersAndAccessGroups(
+        RequestParametersInterface $requestParameters,
+        array $accessGroups
+    ): array {
+        $request
+        = <<<'SQL'
+                SELECT SQL_CALC_FOUND_ROWS DISTINCT
+                m.metric_id, m.metric_name, m.unit_name, CONCAT(r.parent_name, '_', r.name) AS resource_name, r.id as service_id
+                FROM `:dbstg`.`metrics` AS m
+                INNER JOIN `:dbstg`.`index_data` AS id ON id.id = m.index_id
+                INNER JOIN `:dbstg`.`resources` AS r ON r.id = id.service_id
+            SQL;
+
+        $accessGroupIds = array_map(
+            fn ($accessGroup) => $accessGroup->getId(),
+            $accessGroups
+        );
+
+        $request .= ' INNER JOIN `:dbstg`.`centreon_acl` acl
+            ON acl.service_id = r.id
+            AND r.type = 0
+            AND acl.group_id IN (' . implode(',', $accessGroupIds) . ') ';
+
+        $search = $requestParameters->getSearch();
+        $subRequestsInformation = [];
+        if (! empty($search) && array_key_exists('$and', $search)) {
+            $subRequestsInformation = $this->getSubRequestsInformation($search);
+            $request .= $this->buildSubRequestForTags($subRequestsInformation);
+        }
+
+        $request .= <<<'SQL'
+                WHERE r.enabled = 1
+            SQL;
+
+        if (! empty($subRequestsInformation)) {
+            $request .= $subRequestsInformation['service']['request'] ?? '';
+            $request .= $subRequestsInformation['host']['request'] ?? '';
+        }
+
+        $request
+            .= <<<'SQL'
+                    LIMIT 0, 100
+                SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($request));
+        $boundValues = [];
+        if (! empty($subRequestsInformation)) {
+            $boundValues = array_reduce($subRequestsInformation, function ($acc, $subRequestInformation) {
```

Comment:
Adapt with only one array_merge instead of using a reduce
$arrays = [];
foreach(...) {
    $arrays[] = $array;
}
$acc = array_merge(...$arrays);